scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,192 @@
+    censored = np.asarray(censored)
+    if censored.ndim != 1:
+        raise ValueError('censored must be one-dimensional')
+    if (~np.isfinite(x)).any():
+        raise ValueError('x must not contain nan or inf.')
+    if censored.size != x.size:
+        raise ValueError('x and censored must have the same length.')
+    return x, censored.astype(bool)
+
+
+class CensoredData:
+    """
+    Instances of this class represent censored data.
+
+    Instances may be passed to the ``fit`` method of continuous
+    univariate SciPy distributions for MLE fitting.
+
+    Left-, right-, and interval-censored data can be represented.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the ``CensoredData``
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
+    and right-censored data may also be created with the constructor.
+
+    For convenience, the class methods ``left_censored`` and
+    ``right_censored`` are provided to create a ``CensoredData``
+    instance from a single one-dimensional array of observations
+    and a corresponding boolean array to indicate which observations
+    are censored.
+
+    Examples
+    --------
+    In the most general case, a censored data set may contain values that
+    are left-censored, right-censored, interval-censored and not censored.
+    For example, here we create a data set with length five:
+
+    >>> from scipy.stats import CensoredData
+    >>> data = CensoredData([-np.inf, 1, 1.5, 2, 10],
+    ...                     [0,       1, 1.5, 3, np.inf])
+    >>> print(data)
+    CensoredData(5 values: 2 not censored, 1 left-censored,
+    1 right-censored, 1 interval-censored)
+
+    The first value is left-censored; the true (but unknown) value is
+    less than or equal to 0.  The second and third values, 1 and 1.5,
+    are not censored.  The fourth value is interval-censored; the true
+    value is in the interval [2, 3].  The last value is right-censored;
+    the true value is greater than or equal to 10.
+
+    A common case is to have data that is a mix of uncensored values and
+    censored values that are all right-censored (or all left-censored)
+    with the same bound on the censored values.  For example, a measuring
+    device might have an upper limit of 100, so a reading of 100 means
+    the true value is unknown, but the value is not less than 100.
+    Suppose we have nine readings from such a device, with six being
+    [65, 51, 88, 93, 96, 89] and three being 100.  To create an instance
+    of `CensoredData` to represent this data, we can use the class method
+    `CensoredData.right_censored` as follows:
+
+    >>> values = [65, 51, 88, 93, 96, 89, 100, 100, 100]
+    >>> censored = [0, 0, 0, 0, 0, 0, 1, 1, 1]
+
+    A 1 (or any True value) in the sequence ``censored`` indicates
+    that the corresponding value in ``values`` is right-censored.
```

Comment:
Treating integers as boolean is a bit of an anti-pattern, why not just require a boolean array?

Reply:
I was being lazy and taking advantage of the "truthiness" of 0 and 1.  In the new example that replaces this one, I switched to use `False` and `True` explicitly.